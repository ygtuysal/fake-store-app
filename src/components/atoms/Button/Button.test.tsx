import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Button } from '.';

const theme = {
  colors: {
    primary: '#0070f3',
    secondary: '#6c757d',
    text: '#111',
    gray: {
      100: '#f1f3f5',
      400: '#ced4da',
    },
    error: '#dc3545',
    border: '#dee2e6',
    background: '#ffffff',
  },
  borderRadius: {
    md: '4px',
  },
  spacing: {
    xs: '4px',
  },
  transitions: {
    fast: '0.2s',
  },
};

const renderWithTheme = (ui: React.ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

describe('Button', () => {
  it('renders primary button with default styles', () => {
    const { container } = renderWithTheme(<Button>Click me</Button>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.primary);
  });

  it('renders secondary button', () => {
    const { container } = renderWithTheme(<Button variant="secondary">Secondary</Button>);
    expect(container.firstChild).toHaveStyleRule('background-color', theme.colors.secondary);
  });

  it('renders fullWidth button', () => {
    const { container } = renderWithTheme(<Button fullWidth>Full Width</Button>);
    expect(container.firstChild).toHaveStyleRule('width', '100%');
  });

  it('renders hover style for primary button', () => {
    const { container } = renderWithTheme(<Button>Hover me</Button>);
    expect(container.firstChild).toHaveStyleRule('filter', 'brightness(1.1)', {
      modifier: ':hover:not(:disabled)',
    });
  });

  it('renders disabled style', () => {
    const { container } = renderWithTheme(<Button disabled>Disabled</Button>);
    expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed', {
      modifier: ':disabled',
    });
  });
});
