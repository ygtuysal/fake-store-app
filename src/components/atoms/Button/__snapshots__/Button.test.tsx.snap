// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button renders primary button with default styles 1`] = `
.c0 {
  padding: 10px 20px;
  font-size: 16px;
  background-color: #0070f3;
  color: white;
  border: 2px solid transparent;
  width: auto;
  border-radius: 4px;
  font-weight: 600;
  cursor: pointer;
  transition: all 0.2s;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  outline: none;
  position: relative;
  overflow: hidden;
}

.c0:hover:not(:disabled) {
  background-color: #0070f3;
  filter: brightness(1.1);
}

.c0:active:not(:disabled) {
  filter: brightness(0.95);
}

.c0:focus-visible {
  box-shadow: 0 0 0 3px #0070f333;
}

.c0:disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.c0::after {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  width: 0;
  height: 0;
  border-radius: 50%;
  background-color: rgba(255,255,255,0.3);
  transform: translate(-50%,-50%);
  transition: width 0.3s,height 0.3s;
}

.c0:active:not(:disabled)::after {
  width: 300px;
  height: 300px;
}

<button
  class="c0"
>
  Click me
</button>
`;
